packages/core/core/src/public/MutableBundleGraph.js

```diff
@@ -118,8 +118,11 @@ export default class MutableBundleGraph implements IMutableBundleGraph {
       ? assetToInternalAsset(opts.entryAsset)
       : null;
 
+    let target = targetToInternalTarget(opts.target);
     let bundleId = md5FromString(
-      'bundle:' + (opts.uniqueKey ?? nullthrows(entryAsset?.value.id)),
+      'bundle:' +
+        (opts.uniqueKey ?? nullthrows(entryAsset?.value.id)) +
+        target.distDir,
```

Comment:
is there a reason this isn't just a hash of the (ordered) target object?

Reply:
well... I wasn't sure if we wanted to include everything in there, e.g. environment, etc. Two targets in the same dist dir with different environments wouldn't share bundles then maybe?